site/content/features/textblocks.md

```diff
@@ -0,0 +1,444 @@
+    public static void main(String[] args) {
+        String snowmanWithColoredBackground = """
+<pre style="background: skyblue;">
+        \s
+   (")  \s
+  ( : ) \s
+ (     )\s
+---------
+</pre>""";
+        Stream.of(snowmanWithColoredBackground.split("\n")).forEach(line -> System.out.println("|" + line + "|"));
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+### Indent Stripping
+
+Consider a typical text block:
+
+{{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+public class Sandbox {
+    public static void main(String[] args) {
+        String myNameInADiv = """
+<div>
+  Cay 
+</div>
+""";
+        System.out.print(myNameInADiv);
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+Some people prefer to have the text block indented with the rest of the code, so that it doesn't look “out of place”:
+
+{{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+public class Sandbox {
+    public static void main(String[] args) {
+        String myNameInADiv = """
+            <div>
+              Cay 
+            </div>
+            """;
+        System.out.print(myNameInADiv);
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+Then the *common whitespace prefix* is removed, so that the text block consists of the three lines
+
+```
+<div>
+  Cay
+</div>
+```
+
+What about tabs, I hear you cry.
+
+Here are the rules:
+
+1\. A whitespace is any Unicode code point other than `\n` or `\r` for which `Character.isWhitespace` is true. That includes, of course, spaces and tabs as well as 19 other code points. Run this code snippet to find URLs describing them all.
+          {{< sandbox version=java15 preview="true" mainclass="Sandbox" >}}
+{{< sandboxsource "Sandbox.java" >}}
+public class Sandbox {
+    public static void main(String[] args) {
+        for (int i = 0; i < 17 * 65536; i++)
+            if (Character.isWhitespace(i) && i != '\r' && i != '\n')
+                System.out.printf("https://codepoints.net/U+%04X%n", i);
+    }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+I don't know how many programmers use [form feeds](https://codepoints.net/U+000C) or [hair spaces](https://codepoints.net/U+200A) for indenting their code, but Java has them covered.
+
+
+
+2\. Entirely blank lines are not considered in this process, except for the line ending with the closing `"""`.
+
+3\. The indent is the length of the smallest whitespace prefix. In the example above, the indent is 12, as you can see from the following, where white spaces are helpfully replaced with `.`:
+
+```
+        String myNameInADiv = """
+............<div>
+..............Cay 
+............</div>
+............""";
+```
+
+
+
+4\. It doesn't matter what kind of whitespace character is used. Suppose we use a mixture of spaces and tabs (helpfully indicated as ⇥):
+
+```
+        String myNameInADiv = """
+............<div>
+   ⇥   ⇥   ⇥..Cay
+.. 
```

Comment:
This is only 2 white spaces. The text and the example says 3. Or am I missing something?

Reply:
Sorry, my bad. I was miscounting lines. Maybe write "last line" instead of "fifth line" for idiots like me ;)